plugins/GSdx/GSSettingsDlg.cpp

```diff
@@ -282,11 +282,16 @@ bool GSSettingsDlg::OnCommand(HWND hWnd, UINT id, UINT code)
 			theApp.SetConfig("fba", (int)IsDlgButtonChecked(m_hWnd, IDC_FBA));
 			theApp.SetConfig("aa1", (int)IsDlgButtonChecked(m_hWnd, IDC_AA1));
 			theApp.SetConfig("mipmap", (int)IsDlgButtonChecked(m_hWnd, IDC_MIPMAP));
-			theApp.SetConfig("resx", (int)SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0));
-			theApp.SetConfig("resy", (int)SendMessage(GetDlgItem(m_hWnd, IDC_RESY), UDM_GETPOS, 0, 0));
-			theApp.SetConfig("extrathreads", (int)SendMessage(GetDlgItem(m_hWnd, IDC_SWTHREADS), UDM_GETPOS, 0, 0));
 			theApp.SetConfig("accurate_date", (int)IsDlgButtonChecked(m_hWnd, IDC_ACCURATE_DATE));
 			theApp.SetConfig("UserHacks", (int)IsDlgButtonChecked(m_hWnd, IDC_HACKS_ENABLED));
+
+			//Clamp custom resolution and extra rendering threads values to their input range.
+			LPARAM limit_threads = SendMessage(GetDlgItem(m_hWnd, IDC_SWTHREADS), UDM_GETPOS, 0, 0);
+			LPARAM limit_x = SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0);
+			LPARAM limit_y = SendMessage(GetDlgItem(m_hWnd, IDC_RESY), UDM_GETPOS, 0, 0);
+			theApp.SetConfig("extrathreads", LOWORD(limit_threads));
+			theApp.SetConfig("resx", LOWORD(limit_x));
+			theApp.SetConfig("resy", LOWORD(limit_y));
```

Comment:
I don't think `limit_` is good for the variable name. It sort of reads like `Set extrathreads to the thread limit` and so on. It could probably just be
```
theApp.SetConfig("resx", (int16)SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0));
```
or
```
theApp.SetConfig("resx", LOWORD(SendMessage(GetDlgItem(m_hWnd, IDC_RESX), UDM_GETPOS, 0, 0)));
``` 

Anyway, you should explain in the comment that the lower 16 bits returned from a UDM_GETPOS message are automatically restricted to the input range, it isn't obvious.